### PR TITLE
Add UI automation test infrastructure (AST-68408)

### PR DESCRIPTION
**Pull request by elchananarb**
### Description

> 
- Add base test infrastructure for VS automation
- Implement VS instance management
- Add automatic menu navigation
- Setup cleanup mechanisms

### References

> https://checkmarx.atlassian.net/browse/AST-68408
### Testing

> Describe how this change was tested. Be specific about anything not tested and reasons why. If this solution has unit and/or integration testing, tests should be added for new functionality and existing tests should complete without errors.
>
> Please include any manual steps for testing end-to-end or functionality not covered by unit/integration tests.

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR (if applicable).
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used

### CODE REVIEW

File: UITests/BaseTest.cs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+           // Wait until the main window is available
+           _mainWindow = WaitForMainWindow();
+
+           // Handle initial setup
+           SetupVisualStudio().Wait();
+       }
+
+       private static async Task SetupVisualStudio()
+       {
+           await Task.Delay(5000);
```
Comment (greensd4) [inline UITests/BaseTest.cs:35]: we must have this Delays?

Comment (elchananarb) [reply UITests/BaseTest.cs]: I changed the wait in the flow

File: UITests/BaseTest.cs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+            // Wait for launch VS
+            Task.Delay(15000).Wait();
+            _mainWindow = WaitForMainWindow();
+
+           // Handle initial setup
+           SetupVisualStudio().Wait();
+       }
+
+       private static async Task SetupVisualStudio()
```
Comment (miryamfoiferCX) [inline UITests/BaseTest.cs:34]: method is marked as async but is called with .Wait(), which can cause deadlocks. It's better to use await directly if possible

File: UITests/BaseTest.cs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+   [TestClass]
+   public class BaseTest
+   {
+       protected static UIA3Automation _automation;
+       protected static Application _app;
+       protected static AutomationWindow _mainWindow;
+       protected static DTE _dte;
+
+       [ClassInitialize]
+       public static void ClassInitialize(TestContext context)
```
Comment (miryamfoiferCX) [inline UITests/BaseTest.cs:20]: can this method throw any exception during initialize process?
if yes, add try-catch to this logic

File: UITests/BaseTest.cs [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+                   if (window != null && window.IsAvailable)
+                   {
+                       return window;
+                   }
+               }
+               catch (Exception)
+               {
+                   // Ignore exception, retry until timeout
+               }
+               Task.Delay(500).Wait();
```
Comment (miryamfoiferCX) [inline UITests/BaseTest.cs:65]: here also better to use Task.Delay(500).Wait();

File: UITests/CheckmarxTests.cs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+using System.Threading.Tasks;
+
+namespace UITests
+{
+    [TestClass]
+    public class CheckmarxTests : BaseTest
+    {
+        [ClassInitialize]
+        public static void ClassInit(TestContext context)
+        {
```
Comment (miryamfoiferCX) [inline UITests/CheckmarxTests.cs:12]: this method just calls ClassInitialize from the base class, which is a bit redundant. You can directly use the base class's ClassInitialize method without redefining it

**Review by greensd4 — APPROVED** [review]

File: UITests/CheckmarxTests.cs [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+            ClassInitialize(context);
+        }
+
+        [TestMethod]
+        public async Task OpenCheckmarxWindow()
+        {
+            // Find the View menu
+            var viewMenu = _mainWindow.FindFirstDescendant(cf => cf.ByName("View"));
```
Comment (miryamfoiferCX) [inline UITests/CheckmarxTests.cs:20]: did we check somewhere if _mainWindow is not null?

File: UITests/CheckmarxTests.cs [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    public class CheckmarxTests : BaseTest
+    {
+        [ClassInitialize]
+        public static void ClassInit(TestContext context)
+        {
+            ClassInitialize(context);
+        }
+
+        [TestMethod]
+        public async Task OpenCheckmarxWindow()
```
Comment (miryamfoiferCX) [inline UITests/CheckmarxTests.cs:17]: I assume that we can get some exception during this method
no?
if yes - better to add try-catch to this logic

File: UITests/CheckmarxTests.cs [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+        {
+            // Find the View menu
+            var viewMenu = _mainWindow.FindFirstDescendant(cf => cf.ByName("View"));
+            if (viewMenu != null)
+            {
```
Comment (miryamfoiferCX) [inline UITests/CheckmarxTests.cs:22]: maybe break this logic to small method to decrease long nested code

**Pull request merged**